src/java.base/share/classes/java/security/Security.java

```diff
@@ -262,20 +266,37 @@ static void loadInclude(String propFile) {
             }
         }
 
+        private static Path resolve(Path path) {
+            // JDK-8352728: we prefer java.io.File::getCanonicalFile over
+            // java.nio.file.Path::toRealPath because the former is more
+            // fault-tolerant, since the canonical form of a pathname is
+            // specified to exist even for nonexistent/inaccessible files.
+            try {
+                return path.toFile().getCanonicalFile().toPath();
```

Comment:
@seanjmullan If this change is integrated, can we create a follow-up on issue to replace it? I strongly disagree with the changes in this PR, we should not be using File::getCanonicalFile here to work around an issue with an inaccessible parent directory.

Reply:
I think this can be marked as resolved after e80a28ea787b74f22f62f24a09bbf37c8cdbce8a.